src/studio/src/designer/backend.Tests/Controllers/ConvertTextsControllerTests.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Altinn.Studio.Designer.Configuration;
+using Altinn.Studio.Designer.Controllers;
+using Altinn.Studio.Designer.Services.Interfaces;
+using Designer.Tests.Mocks;
+using Designer.Tests.Utils;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using Xunit;
+
+using IRepository = Altinn.Studio.Designer.Services.Interfaces.IRepository;
+
+namespace Designer.Tests.Controllers
+{
+    public class ConvertTextsControllerTests : IClassFixture<WebApplicationFactory<ConvertV1TextsToV2Controller>>
+    {
+        private readonly WebApplicationFactory<ConvertV1TextsToV2Controller> _factory;
+        private readonly string _versionPrefix = "designer/api/v1";
+
+        public ConvertTextsControllerTests(WebApplicationFactory<ConvertV1TextsToV2Controller> factory)
+        {
+            _factory = factory;
+            TestSetupUtils.SetupDirtyHackIfLinux();
+        }
+
+        [Fact]
+        public async Task ConvertTexts_204NoContent()
+        {
+            var targetRepository = Guid.NewGuid().ToString();
+            await TestDataHelper.CopyRepositoryForTest("ttd", "convert-texts", "testUser", targetRepository);
+            HttpClient client = GetTestClient();
+            string dataPathWithData = $"{_versionPrefix}/ttd/{targetRepository}/convertTexts";
+            HttpRequestMessage httpRequestMessage = new HttpRequestMessage(HttpMethod.Put, dataPathWithData);
+            await AuthenticationUtil.AddAuthenticateAndAuthAndXsrFCookieToRequest(client, httpRequestMessage);
+
+            HttpResponseMessage response = await client.SendAsync(httpRequestMessage);
+
+            try
+            {
+                Assert.Equal(StatusCodes.Status204NoContent, (int)response.StatusCode);
+            }
+            finally
+            {
+                System.Threading.Thread.Sleep(100);
+                TestDataHelper.DeleteAppRepository("ttd", targetRepository, "testUser");
+            }
+        }
+
+        private HttpClient GetTestClient()
```

Comment:
See if you can avoid all the mocking, this really shouldn't be necessary. You are already creating a repository for test, and  you have access to the files all the way from the controller. The controller only needs the TextsService which uses the Repository classes which access files. Also see if you can use the pattern that was presented in Techforum by Mirko.